File: ex6_odd_even/src/MutualRecursion.cs
using System;

public class MutualRecursion
{
    private static bool IsOdd(int n) {
        if (n == 0)
            return false;
        return IsEven(n - 1);
    }

    private static bool IsEven(int n) {
        if (n == 0)
            return true;
        return IsOdd(n - 1);
    }

    public static void Main() {
        Console.WriteLine(IsOdd(10));
        Console.WriteLine(IsEven(10));
        Console.WriteLine(IsOdd(9));
        Console.WriteLine(IsEven(9));
        Console.WriteLine(IsOdd(2147483647));
    }
}

File: ex6_odd_even/src/MutualRecursion.java
class MutualRecursion {
    private static boolean isOdd(int n) {
        if (n == 0)
            return false;
        return isEven(n - 1);
    }

    private static boolean isEven(int n) {
        if (n == 0)
            return true;
        return isOdd(n - 1);
    }

    public static void main(String[] args) {
        System.out.println(isOdd(10));
        System.out.println(isEven(10));
        System.out.println(isOdd(9));
        System.out.println(isEven(9));
        System.out.println(isOdd(2147483647));
    }
}
